nomad/state/state_store.go

```diff
@@ -897,6 +887,17 @@ func (s *StateStore) nestedUpdateAllocFromClient(txn *memdb.Txn, watcher watch.I
 	}
 	exist := existing.(*structs.Allocation)
 
+	// Update the job summary before updating the allocation
+	rawJob, err := txn.First("jobs", "id", alloc.JobID)
+	if err != nil {
+		return fmt.Errorf("unable to query job: %v", err)
+	}
+	if rawJob != nil {
```

Comment:
Put the noop logic inside the updateSummaryWithAlloc so that every call site doesn't need this check